src/types/telemetry/telemetryTypes.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { ExtensionContext, ExtensionKind, ExtensionMode, Uri } from 'vscode';
+
+/* eslint-disable header/header */
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+export interface TelemetryReporter {
+  sendTelemetryEvent(
+    eventName: string,
+    properties?: { [key: string]: string },
+    measurements?: { [key: string]: number }
+  ): void;
+
+  sendExceptionEvent(
+    exceptionName: string,
+    exceptionMessage: string,
+    measurements?: { [key: string]: number }
+  ): void;
+
+  dispose(): Promise<void>;
+}
+// end of Copyright (C) Microsoft Corporation. All rights reserved.
+
+export type Measurements = {
+  [key: string]: number;
+};
+
+export type Properties = {
+  [key: string]: string;
+};
+
+export type TelemetryData = {
+  properties?: Properties;
+  measurements?: Measurements;
+};
+
+export type ExtensionInfo = {
+  isActive: boolean;
+  path: string;
+  kind: ExtensionKind;
+  uri: Uri;
+  loadStartDate: Date;
+};
+
+export type ExtensionsInfo = {
+  [extensionId: string]: ExtensionInfo;
+};
+
+export type ActivationInfo = Partial<ExtensionInfo> & {
+  startActivateHrTime: [number, number];
+  activateStartDate: Date;
+  activateEndDate?: Date;
+  extensionActivationTime: number;
+  markEndTime?: number;
+};
+
+export interface TelemetryServiceInterface {
+  /**
+   * Initialize Telemetry Service during extension activation.
+   * @param extensionContext extension context
+   */
+  initializeService(extensionContext: ExtensionContext): Promise<void>;
+  /**
+   * Initialize Telemetry Service with name, foo, version, and extensionMode.
```

Comment:
Did you mean `apiKey` instead of `foo`?

```suggestion
   * Initialize Telemetry Service with name, apiKey, version, and extensionMode.
```